src/plugins/intel_gpu/src/kernel_selector/tensor_type.cpp

```diff
@@ -550,6 +550,27 @@ DataTensor DataTensor::FlattenEverything() const {
     return res;
 }
 
+DataTensor DataTensor::SwapXYSpatials(bool swapXY = false) const {
```

Comment:
This interface looks weird to me. If swap is not needed, then SwapXYSpatials shouldn't be called at all. I'd suggest moving swap logic from get_default_params() common method to convolution_impl:
```cpp
     bool can_swap_xy = conv_params.inputs[0].Batch().v == 1 && primitive->deformable_mode == false
            && conv_params.inputs[0].X().v == 1 && conv_params.inputs[0].Y().v > 1
            && conv_params.outputs[0].X().v == 1 && conv_params.outputs[0].Y().v > 1
            && conv_params.weights.X().v == 1 && conv_params.inputs[0].X().pad.Total() == 0
            && conv_params.inputs[0].Dimentions() == 4);
     if (can_swap_xy) {
            conv_params.inputs[0].SwapXY(); 
            conv_params.outputs[0].SwapXY(); 
            conv_params.weights.SwapXY(); 
            conv_params.filterSize = { ky, kx, kz };
            conv_params.padding = {pad_y, pad_x, pad_z};
            conv_params.stride = {stride_y, stride_x, stride_z};
            conv_params.dilation = {dilation_y, dilation_x, dilation_z};
     }

```

Reply:
It looks weird to me too. I don't have to worry about fused-op?

Reply:
You have to, but it can be done in this place too. No need to modify common function to my mind.

Reply:
Fixed